### PR TITLE
[RLC] add new flag multi-client

### PR DESCRIPTION
**Pull request by dw511214992**
This PR adds a new flag `multi-client` to represents it's generating multi clients.

Previously, codegen generates multi clients when `batch` task is set. However, if we only want to use `batch` task to generate multi packages, it's no way. So we need a new flag `multi-client` to indicate codegen is generating multi client.

### CODE REVIEW

File: src/generators/indexGenerator.ts
Comment (dw511214992) [reply src/generators/indexGenerator.ts]: I have add a check for both `multi-client` and `batch`.

**Comment by dw511214992** [comment]
> Thanks for sending this out @dw511214992! I like the idea of adding a multi-client flag to make the intent explicit. Quick question, what happens if there is a `multi-client` flag set but there is no `batch`? How would the generator handle that scenario?
> 
> Changes look good in general, I'd like @qiaozha to also take a look

I have added a check for both `multi-clent` and `batch`, and user must set `multi-client` flag and `batch` to generate multi client.

**Review by joheredi — APPROVED** [review]
Looks good to me. Please get @qiaozha's approval before merging

**Review by qiaozha — APPROVED** [review]

**Pull request merged**
